MaxText/layers/attentions.py

```diff
@@ -1215,19 +1244,43 @@ def out_projection(self, output_dim: int, out: Array) -> Array:
     return out_proj
 
   def apply_rotary_embedding(self, inputs: Array, inputs_positions: Array, name: str):
-    if self.config.model_name.startswith("llama3.1"):
+    """Applies rotary embeddings, handling different model types.
+
+    Args:
+      inputs: The input tensor to apply rotary embeddings to.
+      inputs_positions: The positions of the inputs.
+      name: A name for the embedding layer.
+
+    Returns:
+      The input tensor with rotary embeddings applied.
+    """
+    if self.config.attention_type == AttentionType.MLA.value:
+      # For MLA attention RoPE is applied to only `self.qk_rope_head_dim` portion the heads.
+      rope_embedding_dims = self.qk_rope_head_dim
+    else:
+      rope_embedding_dims = self.head_dim
+
+    rope_type = self.config.rope_type.lower()
```

Comment:
Should we add a validation in [pyconfig file](https://github.com/AI-Hypercomputer/maxtext/blob/10b24ab5d56aa128d8b5a22ea185e4115b70860c/MaxText/pyconfig.py#L4)? i.e. rope type can only be one of "default", "llama3.1" or "yarn"

Reply:
Done